horace_core/sqw/file_io/@sqw_binfile_common/put_pix.m

```diff
@@ -167,6 +167,7 @@
     npix_to_write = npix_hi-npix_lo+1;
```

Comment:
The if/else here is switching on block not page-size so the else block may need paging too,

Two specific issues:
- are the start and end pixels whole dataset index or in-page pixel indexes, and does that match the expected arguments for `get_pixels()`
- if the chunk of pixels being written is larger than a single page this block/page loop will need to be rewritten to handle looping both

Reply:
As noted in the opening comment: "Cases where write_all is false or the pixel range is limited have still to be done. " But thanks for the specifics. Note from Harry in #389 "There is a conversation that needs to be had about whether that range should index into the current page of pixels, or into the full pixel array, and what the implications of both options are." I have pursued this in the debugger but am still unclear about the use-case.